addons/mail/models/mail_channel.py

```diff
@@ -679,6 +679,7 @@ def channel_get(self, partners_to, pin=True):
             })
             # broadcast the channel header to the other partner (not me)
             channel._broadcast(partners_to)
+        channel._broadcast(self.env.user.partner_id.ids)
```

Comment:
This added broadcast should only be on the "if". The "else" broadcast also includes the current partner (comment is obsolete, current partner is added to `partners_to` at the top of the method).